### DISCUSSION TITLE
Is it possible to use Benthos as a transport layer only

### DISCUSSION THREAD

**Discussion by tukusejssirs** [redpanda-data/connect#1664] [Q&A]
I have learnt about Benthos only recently. I haven’t actually used it (yet), but I like it quite much, esp the delivery guarantees.

However, I prefer using [Nest](https://nestjs.com/) as my go-to backend framework. I was thinking about using Benthos as microservice transport (using event-based messaging style to execute some functions), however, I have no idea if it is possible.

From what I’ve read, it seems like Benthos is quite monolithic, therefore there is no separate transport layer (like TCP, KAFKA). However, if it is still possible, I’d like to create a [custom transporter](https://docs.nestjs.com/microservices/custom-transport).

You might ask why. Well, the thing is that Benthos, while it is awesome, is configured (let’s say) at _start time_. I have a use case where I want to use change the config at runtime without restarting Benthos itself. I understand that Benthos is _stateless_, but the config seems to be static/non-changable after Benthos starts.

**Comment by mihaitodor** [comment]
Oops, I meant to reply to this sooner @tukusejssirs. Benthos does have [Streams Mode](https://www.benthos.dev/docs/guides/streams_mode/about) which comes with a [REST API](https://www.benthos.dev/docs/guides/streams_mode/using_rest_api). Hope this helps.
